python/lsst/pipe/base/graphBuilder.py

```diff
@@ -784,12 +784,13 @@ def connectDataIds(
                         ref = refsForRow[datasetType.name]
                         quantum.outputs[datasetType.name][ref.dataId] = ref
             if n < 0:
+                _LOG.fatal("Initial data ID query returned no rows, so QuantumGraph will be empty.")
                 emptiness_explained = False
                 for message in commonDataIds.explain_no_results():
-                    _LOG.warning(message)
+                    _LOG.fatal(message)
```

Comment:
Quick comment, I know it's not in review yet, but Python really wants you to use `_LOG.critical()` here:

> logging.fatal = fatal(msg, *args, **kwargs)
>    Don't use this function, use critical() instead.